geoutils/georaster.py

```diff
@@ -85,8 +90,25 @@ class Raster(object):
     # This only gets set if a disk-based file is read in.
     # If the Raster is created with from_array, from_mem etc, this stays as None.
     filename = None
-    _is_modified = None
-    _disk_hash = None
+    _is_modified: Optional[bool] = None
+    _disk_hash: Optional[int] = None
+
+    # Rasterio-inherited names and types are defined here to get proper type hints.
```

Comment:
Oh, that seems annoying for future maintenance since it is more or less a duplicate of default_attrs...
Isn't there a way to combine both? Could we replace the list by a dictionary containing the type?

Reply:
It was hard, but I did it! The `default_attrs` is commented out and is instead replaced by type annotations in the class definition, combined with a new `self._get_rio_attrs()`.

`self._get_rio_attrs()` finds all attributes that have been annotated that corresponds to attributes in the DatasetReader (transform, bounds, res, etc.) and sets the attributes accordingly. This means that as long as the annotations are up to date, everything works well.

Reply:
I think that this is necessary for pylint. I also had issues with those attributes in PyCharm (not recognized), because those were not initialized but only defined by `setattr(self)`. It should be fine, hopefully we won't add a new attribute every day!

Reply:
Does it mean that now all of rasterio's attributes are copied? Or still the same as before?

Reply:
It's exactly the same as before. It's just that the common attributes are set in a different place (in the class instead of in the list)